### PR TITLE
modified test for IndexRoute and IndexLink nesting

### PR DESCRIPTION
**Pull request by pjuke**
When nesting IndexRoute's, some of them become invalidly marked as active, this PR contains the failing test for the case.

### CODE REVIEW

**Comment by iNikNik** [comment]
It seems here https://github.com/pjuke/react-router/blob/IndexRoute-test/modules/__tests__/IndexLink-test.js#L137 should be ```.toEqual('')```. Because productsLink is IndexLink to ```/website/products``` and route ```/website/products/15``` is not index.

**Comment by pjuke** [comment]
Yeah you are right, that was a typo from my side. It should be a `<Link>` instead.
However that is not the problem. The `when linking to the contact`-test still fails for example; since the `overview` page still is active.

**Comment by iNikNik** [comment]
@pjuke check out my PR, I hope I fixed this)

**Comment by mjackson** [comment]
I believe this was fixed in 37d9bacd432f1b588df8b1f72227383d1a9f3197

File: modules/__tests__/IndexLink-test.js [hunk 1]
```diff
@@ -54,25 +85,57 @@ describe('An <IndexLink>', function () {
...
+      })
+    })
+  })
+
+  describe('when linking to the contact', function () {
+    it("is active and other routes are not", function (done) {
+      React.render((
+        <Router history={createHistory('/website/contact')} routes={routes} />
+      ), node, function () {
+        expect(node.querySelector('#overviewLink').className).toEqual('')
```
Comment (mjackson) [inline modules/__tests__/IndexLink-test.js]: This expectation is not accurate. At `/website/contact` both the `Website` route *and* the `WebsiteContact` routes are active.

Comment (mjackson) [inline modules/__tests__/IndexLink-test.js]: Oh wait, `#overviewLink` is an `IndexLink`. Ok, you're right.

**Comment by mjackson** [comment]
Rebased and merge manually in 2732586a53fa0d81a25902b685038336d815879a

**Pull request closed without merging**